mentat/code_feature.py

```diff
@@ -99,42 +98,39 @@ class CodeFeature:
     Includes a section of code and an annotation method.
 
     Attributes:
-        path: The absolute path to the file.
+        path: A path to a file.
         intervals: The lines in the file.
         level: The level of information to include.
         diff: The diff annotations to include.
     """
 
     def __init__(
         self,
-        path: str | Path,
+        path: str | Path,  # TODO: only allow Path type
         level: CodeMessageLevel = CodeMessageLevel.CODE,
         diff: str | None = None,
         user_included: bool = False,
         name: Optional[str] = None,
     ):
-        if Path(path).exists():
-            self.path = Path(path)
-            self.intervals = [Interval(0, math.inf)]
+        ctx = SESSION_CONTEXT.get()
+
+        validated_path = validate_and_format_path(path=Path(path), cwd=ctx.cwd, check_for_text=False)
+        if ":" in str(validated_path):
```

Comment:
This isn't safe. Paths are allowed to have colons in them. And in the case of windows they almost always do.

Reply:
Will update this